recipes/fast-dds/3.x/conanfile.py

```diff
@@ -0,0 +1,156 @@
+import os
+import textwrap
+
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.files import (
+    apply_conandata_patches,
+    collect_libs,
+    copy,
+    export_conandata_patches,
+    get,
+    rename,
+    rm,
+    rmdir,
+    save,
+)
+from conan.tools.microsoft import check_min_vs, is_msvc_static_runtime, is_msvc, msvc_runtime_flag
+from conan.tools.scm import Version
+
+
+required_conan_version = ">=2.0"
+
+
+class FastDDSConan(ConanFile):
+    name = "fast-dds"
+    license = "Apache-2.0"
+    homepage = "https://fast-dds.docs.eprosima.com/"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "The most complete OSS DDS implementation for embedded systems."
+    topics = ("dds", "middleware", "ipc")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_ssl": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_ssl": False,
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 11
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "11",
+            "clang": "15",
+            "apple-clang": "15",
```

Comment:
The required compiler versions should be relaxed. The only requirement is C++11 

https://github.com/conan-io/conan-center-index/pull/27887#discussion_r2194785573